File: PooCC01CNetz/Semana02/Python/Ejercicios4/MetodosSobrecarga.py
# En python no se puede hacer sobrecarga de metodos ya que solo se conservara la ultima definicion de suma()
# Por eso mismo aquí le cambio el nombre a suma2 para dos valores y suma3 para dos valores
class MetodosSobrecarga:
    def main(self):
        ms = MetodosSobrecarga()
        res = ms.suma2(3, 5)
        print(res)
        res = ms.suma3(4, 7, 9)
        print(res)
        resf = ms.suma2(8, 9.9)
        print(resf)

    def suma2(self, a, b):
        res = a + b
        return res

    def suma3(self, a, b, c):
        res = a + b + c
        return res

# Crear una instancia de la clase y llamo al método main
ms = MetodosSobrecarga()
ms.main()

File: PooCC01CNetz/Semana02/Java/Ejercicios4/MetodosSobrecarga.java
public class MetodosSobrecarga {
    public static void main(String[] args) throws Exception {
        // Declaro el objeto de la misma clase
        MetodosSobrecarga ms = new MetodosSobrecarga();
        
        // Llamo a metodos de la misma clase
        int res = ms.suma(3, 5);
        System.out.println(res);
        res = ms.suma(4, 7, 9);
        System.out.println(res);
        double resf = ms.suma(8, 9.9);
        System.out.println(resf);
    }

    // Creo el metodo suma con dos variables en los parametros(dos int)
    public int suma(int a, int b) {
        int res = a + b;
        return res;
    }

    // Creo el metodo suma con tres variables en los parametros
    public int suma(int a, int b, int c) {
        int res = a + b + c;
        return res;
    }

    // Creo el metodo suma con dos variables en los parametros(uno int y otro double)
    public double suma(int a, double d) {
        double res = a + d;
        return res;
    }
}